JsonSchema.Generation/Generators/ObjectSchemaGenerator.cs

```diff
@@ -68,10 +68,7 @@ public void AddConstraints(SchemaGenerationContextBase context)
 
 			var memberContext = SchemaGenerationContextCache.Get(member.GetMemberType(), unconditionalAttributes);
 
-			var name = SchemaGeneratorConfiguration.Current.PropertyNamingMethod(member.Name);
-			var nameAttribute = unconditionalAttributes.OfType<JsonPropertyNameAttribute>().FirstOrDefault();
-			if (nameAttribute != null)
-				name = nameAttribute.Name;
```

Comment:
As stated before, this is the preferred behavior.  `[JsonPropertyName]` should override this setting, not be a part of it.

Reply:
I tried to implement it the way you wanted.